dohproxy/server_protocol.py

```diff
@@ -60,11 +124,50 @@ def datagram_received(self, data, addr):
                 interval
             )
         )
-        self.queue.put_nowait(dnsr)
+        self.fut.set_result(dnsr)
+
+
+class DNSClientProtocolUDP(DNSClientProtocol):
+
+    def connection_made(self, transport):
+        self.send_helper(transport)
+        self.transport.sendto(self.dnsq.to_wire())
+
+    def datagram_received(self, data, addr):
+        dnsr = dns.message.from_wire(data)
+        self.receive_helper(dnsr)
         self.transport.close()
 
     def error_received(self, exc):
-        print('Error received:', exc)
+        self.logger.debug('Error received')
```

Comment:
you should print the error that was received. As is, the message would not be informative.